### PR TITLE
FS-004: add tests with React Testing Library.

### PR DESCRIPTION
**Pull request by MykhailoForge**
Add tests due to https://github.com/MykhailoForge/prep-proj/issues/4

### CODE REVIEW

**Comment by MykhailoForge** [comment]
photo_2023-06-07_13-50-50

File: src/modules/language-selector/LanguageSelector.test.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+describe("languageSelector test", () => {
+  test("Should render component", () => {
+    render(
+      <Provider store={store}>
+        <LanguageSelector changeHandler={mockChangeFunc} />
+      </Provider>
+    );
+  });
+
+  test("Should handle select", () => {
```
Comment (stanislavSoro) [inline src/modules/language-selector/LanguageSelector.test.tsx]: How do you can be sure that it's worked?

File: src/modules/queue-visualizer/QueueVlisualizer.test.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      <Provider store={store}>
+        <QueueVisualizer />
+      </Provider>,
+      { wrapper: BrowserRouter }
+    );
+
+    expect(screen.getByTestId("back-button")).toBeInTheDocument();
+  });
+
+  test("Should process input and add button", () => {
```
Comment (stanislavSoro) [inline src/modules/queue-visualizer/QueueVlisualizer.test.tsx]: You shouldn't only test clicks and changes. You need to check that actual expected result is presented on the screen.

File: src/modules/core/AppRouter.tsx [hunk 3]
```diff
@@ -5,11 +5,22 @@ import { QUEUE_VISUALIZER, ROOT, STACK_VISUALIZER } from "./routes.ts";
...
+import { useAppDispatch } from "./store/store.ts";
+import { setLanguage } from "./store/coreSlice.ts";
+import { useTranslation } from "react-i18next";
 
 function AppRouter() {
+  const dispatch = useAppDispatch();
+  const { i18n } = useTranslation();
+
+  const handleLanguageChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(setLanguage(event.target.value));
```
Comment (NickForge) [inline src/modules/core/AppRouter.tsx]: you can add 
`const language = event.target.value`
and use it

File: src/modules/language-selector/LanguageSelector.tsx [hunk 4]
```diff
@@ -1,44 +1,35 @@
...
-          {languageArr.map((langItem) => (
-            <MenuItem value={langItem.item} key={langItem.id}>
-              {langItem.item}
-            </MenuItem>
-          ))}
-        </Select>
-      </FormControl>
+      <TextField
+        select
+        inputProps={{ "data-testid": "select" }}
```
Comment (NickForge) [inline src/modules/language-selector/LanguageSelector.tsx]: maybe call it more specific? some alike `language-selector`

File: src/modules/theme/components/DataListLayout.tsx [hunk 5]
```diff
@@ -53,12 +54,14 @@ export default function DataListLayout({
...
             onClick={firstButtonHandler}
+            data-testid="add-button"
           >
             {firstButtonName}
           </DataVisualizeControl>
           <DataVisualizeControl
             variant="contained"
             onClick={secondButtonHandler}
+            data-testid="remove-button"
           >
```
Comment (NickForge) [inline src/modules/theme/components/DataListLayout.tsx:65]: please use more specific `data-testid`

File: src/modules/language-selector/LanguageSelector.tsx (hunk 4 above)
Comment (MykhailoForge) [reply src/modules/language-selector/LanguageSelector.tsx]: removed it as unnecessary.

File: src/modules/queue-visualizer/QueueVisualizer.test.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  test("Should process input and add button", () => {
+    render(
+      <Provider store={store}>
+        <QueueVisualizer />
+      </Provider>,
+      { wrapper: BrowserRouter }
+    );
+
+    const textInput = screen.getByTestId("data-list-text-input");
+    fireEvent.change(textInput, { target: { value: "queue elem" } });
```
Comment (stanislavSoro) [inline src/modules/queue-visualizer/QueueVisualizer.test.tsx]: String values should be a const(!). You are using this value in 2 places.

File: src/modules/queue-visualizer/QueueVisualizer.test.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+    const addButton = screen.getByTestId("data-list-add-button");
+    fireEvent.click(addButton);
+
+    expect(screen.getByText(/queue elem/i)).toBeInTheDocument();
```
Comment (stanislavSoro) [inline src/modules/queue-visualizer/QueueVisualizer.test.tsx]: You are parsing the whole screen to find the value in the specific place. But you can add a data-testid for the block where it should be presented and find it there.

File: src/modules/queue-visualizer/QueueVisualizer.test.tsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        <QueueVisualizer />
+      </Provider>,
+      { wrapper: BrowserRouter }
+    );
+
+    const removeButton = screen.getByTestId("data-list-remove-button");
+    fireEvent.click(removeButton);
+
+    await waitFor(() => {
+      expect(screen.queryByText(/1/i)).not.toBeInTheDocument();
```
Comment (stanislavSoro) [inline src/modules/queue-visualizer/QueueVisualizer.test.tsx]: You don't need to search the whole screen for the specific value.

File: src/modules/stack-visualizer/StackVisualizer.test.tsx [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  test("Should process input and add button", () => {
+    render(
+      <Provider store={store}>
+        <StackVisualizer />
+      </Provider>,
+      { wrapper: BrowserRouter }
+    );
+
+    const textInput = screen.getByTestId("data-list-text-input");
+    fireEvent.change(textInput, { target: { value: "stack elem" } });
```
Comment (stanislavSoro) [inline src/modules/stack-visualizer/StackVisualizer.test.tsx]: string should be a const

File: src/modules/language-selector/LanguageSelector.test.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  test("Should handle select", async () => {
+    const { container } = render(
+      <Provider store={store}>
+        <LanguageSelector changeHandler={mockChangeFunc} />
+      </Provider>
+    );
+
+    await waitFor(() => userEvent.click(getByRole(container, "button")));
+    await waitFor(() => userEvent.click(screen.getByText("ua")));
+    screen.debug();
```
Comment (NickForge) [inline src/modules/language-selector/LanguageSelector.test.tsx]: please clean up your code before make commit/push

**Review by stanislavSoro — APPROVED** [review]

**Pull request merged**